core: engine: metavariable-pattern: Fix location info

IagoAbal opened the pull request:
File, file content, and Generic AST, must all have consistent location
info. This was being done when using the `language` sub-key but not
otherwise, yet it is still needed e.g. when using `pattern-regex`.

Closes #3503

test plan:
make test # tests included



PR checklist:
- [x] documentation is up to date
- [x] changelog is up to date

aryx commented on semgrep-core/src/core/ast/Map_AST.ml:
```diff
@@ -1099,3 +1099,21 @@ let (mk_visitor : visitor_in -> visitor_out) =
+          k
+            {
+              s with
+              s_range = Common.map_opt (fun (x, y) -> (fix x, fix y)) s.s_range;
```
good call, I would have forgotten that one.

IagoAbal replied:
I forgot too, but the tests reminded me about it!

aryx commented on semgrep-core/src/engine/Match_rules.ml:
```diff
@@ -647,7 +647,13 @@ and satisfies_metavar_pattern_condition env r mvar opt_xlang formula =
         mvar;
       false
   | Some mval -> (
+      (* We will crete a temporary file with the content of the metavariable,
```
typo, crete -> create

IagoAbal committed:
core: engine: metavariable-pattern: Fix location info

File, file content, and Generic AST, must all have consistent location
info. This was being done when using the `language` sub-key but not
otherwise, yet it is still needed e.g. when using `pattern-regex`.

Closes #3503

test plan:
make test # tests included
